exchanges/bitfinex/bitfinex_wrapper.go

```diff
@@ -732,35 +736,75 @@ func (b *Bitfinex) GetOrderInfo(ctx context.Context, orderID string, pair curren
 }
 
 // GetDepositAddress returns a deposit address for a specified currency
-func (b *Bitfinex) GetDepositAddress(ctx context.Context, c currency.Code, accountID string) (string, error) {
+func (b *Bitfinex) GetDepositAddress(ctx context.Context, c currency.Code, accountID, chain string) (*deposit.Address, error) {
 	if accountID == "" {
-		accountID = "deposit"
+		accountID = "funding"
 	}
 
-	method, err := b.ConvertSymbolToDepositMethod(ctx, c)
-	if err != nil {
-		return "", err
+	if c == currency.USDT {
+		c = currency.NewCode("UST")
+	}
+
+	if err := b.PopulateAcceptableMethods(ctx); err != nil {
+		return nil, err
+	}
+
+	methods := AcceptableMethods.Lookup(c)
+	if len(methods) == 0 {
+		return nil, errors.New("unsupported currency")
+	}
+	method := methods[0]
+	if len(methods) > 1 && chain != "" {
+		method = chain
+	} else if len(methods) > 1 && chain == "" {
+		return nil, fmt.Errorf("a chain must be specified, %s available", methods)
 	}
 
 	resp, err := b.NewDeposit(ctx, method, accountID, 0)
-	return resp.Address, err
+	if err != nil {
+		return nil, err
+	}
+	return &deposit.Address{
+		Address: resp.Address,
+		Tag:     resp.PoolAddress,
+	}, err
 }
 
 // WithdrawCryptocurrencyFunds returns a withdrawal ID when a withdrawal is submitted
 func (b *Bitfinex) WithdrawCryptocurrencyFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
 	if err := withdrawRequest.Validate(); err != nil {
 		return nil, err
 	}
+
+	if err := b.PopulateAcceptableMethods(ctx); err != nil {
+		return nil, err
+	}
+
+	if withdrawRequest.Currency == currency.USDT {
+		withdrawRequest.Currency = currency.NewCode("UST")
+	}
+
+	methods := AcceptableMethods.Lookup(withdrawRequest.Currency)
+	if len(methods) == 0 {
+		return nil, errors.New("unsupported currency")
+	}
+	method := methods[0]
+	if len(methods) > 1 && withdrawRequest.Crypto.Chain != "" {
+		method = withdrawRequest.Crypto.Chain
+	} else if len(methods) > 1 && withdrawRequest.Crypto.Chain == "" {
+		return nil, fmt.Errorf("a chain must be specified, %s available", methods)
+	}
```

Comment:
Made a slight change to do a comparison for the supplied chain vs avail, this has to support cases where:

a) A coin like BTC which will return just a BITCOIN method with or without a chain specified
b) USDT which returns [TETHERUSDTHEZ TETHERUSS TETHERUSDTOMG TETHERUSO TETHERUSE TETHERUSDTSOL TETHERUSL TETHERUSDTALG TETHERUSX TETHERUSDTBCH TETHERUSDTDVF] methods. The change now ensures that it's found in the avail list if specified, otherwise returns an error if no chain is set for it